src/components/SignUp.js

```diff
@@ -0,0 +1,130 @@
+import { useState } from "react";
+import { validateEmail } from "../utilis/validateEmail";
+
+const PasswordErrorMessage = () => {
+  return (
+    <p className="text-xs text-red-600 m-1">
+      The password should contain 8 characters
+    </p>
+  );
+};
+
+const SignUp = () => {
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState({ value: "", isTouched: false });
+
+  const handleChange = (e) => {
+    if (e.target.name === "firstName") {
+      setFirstName(e.target.value);
+    } else if (e.target.name === "lastName") {
+      setLastName(e.target.value);
+    } else if (e.target.name === "email") {
+      setEmail(e.target.value);
+    } else if (e.target.name === "password") {
+      setPassword({ ...password, value: e.target.value });
+    } else {
+      console.log("nothing"); // fix this
+    }
```

Comment:
@Usman-Ubaid , instead of this long chain of `if-else`, we need to use `switch` statement.